unit/solvers/strings/string_format_builtin_function/length_for_format_specifier.cpp

```diff
@@ -0,0 +1,261 @@
+#include <util/string_expr.h>
+/*******************************************************************\
+
+Module: Unit tests for length_for_format_specifier
+        solvers/refinement/string_format_builtin_function.cpp
+
+Author: Diffblue Ltd.
+
+\*******************************************************************/
+
+#include <testing-utils/use_catch.h>
+
+#include <solvers/strings/format_specifier.h>
+#include <solvers/strings/string_format_builtin_function.h>
+#include <util/simplify_expr.h>
+#include <util/symbol_table.h>
+
+// Create array_string_exprt from array of characters
+array_string_exprt from_char_vector(
+  std::vector<mp_integer> string_array,
+  const typet &char_type,
+  const typet &length_type,
+  const pointer_typet &pointer_type,
+  array_poolt &array_pool)
+{
+  const array_typet array_type(
+    char_type, from_integer(string_array.size(), length_type));
+  std::vector<exprt> expr_array;
+  expr_array.reserve(string_array.size());
+  std::transform(
+    string_array.begin(),
+    string_array.end(),
+    std::back_inserter(expr_array),
+    [&](const mp_integer el) { return from_integer(el, char_type); });
```

Comment:
:pick: could be simpler with `const auto expr_array = make_range(string_array).map([&](const mp_integer el) { return from_integer(el, char_type); }).collect<std::vector<exprt>>()`

Reply:
I'll skip that one.